src/Entity/FileEntity.php

```diff
@@ -14,11 +14,33 @@
 use Drupal\Core\StreamWrapper\StreamWrapperInterface;
 use Drupal\file\Entity\File;
 use Drupal\file\FileInterface;
+use Drupal\file_entity\FileEntityInterface;
 
 /**
  * Replace for the core file entity class.
  */
-class FileEntity extends File {
+class FileEntity extends File implements FileEntityInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getMetadata($property) {
+    return $this->metadata[$property];
```

Comment:
Make sure to do an isset() check here to avoid notices if a certain metadata is not set. Maybe also add a hasMetadata() method?